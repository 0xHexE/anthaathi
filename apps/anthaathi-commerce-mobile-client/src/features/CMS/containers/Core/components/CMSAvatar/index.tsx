import {View} from 'react-native';
import React from 'react';
import {Avatar as Avatar} from 'react-native-paper';
import {CoreComponentType} from '../../../../types/common';

export interface AvatarProps {
  userData: {
    name: string;
    photo: string;
  };
  size?: number;
}

const CMSAvatar = ({userData, size = 120}: AvatarProps) => {
  return (
    <View
      testID="CMSAvatar"
      style={{alignItems: 'center', marginTop: 10, marginBottom: 15}}>
      {userData.photo ? (
        <Avatar.Image
          size={size}
          source={{
            uri: userData.photo,
          }}
        />
      ) : (
        <Avatar.Text size={size} label={userData.name} />
      )}
    </View>
  );
};

export default CMSAvatar;

export const AvatarCMSInput = {
  _component: CoreComponentType.CMSAvatar,
  component: CMSAvatar,
};
