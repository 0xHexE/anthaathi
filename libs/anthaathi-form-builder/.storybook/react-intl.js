const locales = ['en-US'];

const messages = locales.reduce(
  (acc, lang) => ({
    ...acc,
    [lang]: import(`../locales/compiled-locales/${lang}.json`), // whatever the relative path to your messages json is
  }),
  {},
);

const formats = {};

export const reactIntl = {
  defaultLocale: 'en-US',
  locales,
  messages,
  formats,
};
