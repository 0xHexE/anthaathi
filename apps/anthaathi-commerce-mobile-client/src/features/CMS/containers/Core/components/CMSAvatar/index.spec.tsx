import React from 'react';
import 'react-native';
import {render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import CMSAvatar from './index';

describe('CMSAvatar', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <CMSAvatar userData={{name: 'test', photo: 'src'}} />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('CMSAvatar')).toBeTruthy();
  });
});
