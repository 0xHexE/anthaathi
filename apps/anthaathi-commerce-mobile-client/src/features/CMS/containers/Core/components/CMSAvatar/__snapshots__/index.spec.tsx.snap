// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CMSAvatar should render the item 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "marginBottom": 15,
      "marginTop": 10,
    }
  }
  testID="CMSAvatar"
>
  <View
    style={
      [
        {
          "backgroundColor": "#6200ee",
          "borderRadius": 60,
          "height": 120,
          "width": 120,
        },
        undefined,
      ]
    }
  >
    <Image
      source={
        {
          "uri": "src",
        }
      }
      style={
        {
          "borderRadius": 60,
          "height": 120,
          "width": 120,
        }
      }
    />
  </View>
</View>
`;
